File: app/src/main/java/com/letianpai/robot/control/callback/ControlSteeringEngineCallback.java
package com.letianpai.robot.control.callback;

/**
 * @author liujunbin
 */
public class ControlSteeringEngineCallback {

    private ControlSteeringEngineListener mControlSteeringEngineListener;


    private static class ControlSteeringEngineCallbackHolder {
        private static final ControlSteeringEngineCallback instance = new ControlSteeringEngineCallback();
    }

    private ControlSteeringEngineCallback() {

    }

    public static ControlSteeringEngineCallback getInstance() {
        return ControlSteeringEngineCallbackHolder.instance;
    }


    public interface ControlSteeringEngineListener {
        void onControlSteeringEngine(boolean footSwitch, boolean sensorSwitch);
    }


    public void setControlSteeringEngineListener (ControlSteeringEngineListener listener) {
        this.mControlSteeringEngineListener = listener;
    }



    public void setControlSteeringEngine(boolean footSwitch, boolean sensorSwitch) {
        mControlSteeringEngineListener.onControlSteeringEngine(footSwitch,sensorSwitch);
    }



}

File: app/src/main/java/com/letianpai/robot/control/callback/ControlSteeringEngineCallback.kt
package com.letianpai.robot.control.callback

/**
 * @author liujunbin
 */
class ControlSteeringEngineCallback private constructor() {
    private var mControlSteeringEngineListener: ControlSteeringEngineListener? = null


    private object ControlSteeringEngineCallbackHolder {
        val instance: ControlSteeringEngineCallback = ControlSteeringEngineCallback()
    }

    interface ControlSteeringEngineListener {
        fun onControlSteeringEngine(footSwitch: Boolean, sensorSwitch: Boolean)
    }


    fun setControlSteeringEngineListener(listener: ControlSteeringEngineListener?) {
        this.mControlSteeringEngineListener = listener
    }


    fun setControlSteeringEngine(footSwitch: Boolean, sensorSwitch: Boolean) {
        mControlSteeringEngineListener!!.onControlSteeringEngine(footSwitch, sensorSwitch)
    }


    companion object {
        val instance: ControlSteeringEngineCallback
            get() = ControlSteeringEngineCallbackHolder.instance
    }
}
